### PR TITLE
JAVA-539 cse.request.timeout support dynamic refresh

### PR DESCRIPTION
**Pull request by laijianbin**

### CODE REVIEW

File: transports/transport-highway/src/main/java/io/servicecomb/transport/highway/HighwayClient.java [hunk 1]
```diff
@@ -62,7 +66,13 @@ public void init(Vertx vertx) throws Exception {
 
   private TcpClientConfig createTcpClientConfig() {
     TcpClientConfig tcpClientConfig = new TcpClientConfig();
-    tcpClientConfig.setRequestTimeoutMillis(AbstractTransport.getRequestTimeout());
+    DynamicLongProperty prop = AbstractTransport.getRequestTimeoutProperty();
+    prop.addCallback(new Runnable(){
```
Comment (WillemJiang) [inline transports/transport-highway/src/main/java/io/servicecomb/transport/highway/HighwayClient.java]: It's better to add a comment on this line to support setup the request timeout dynamically.

File: core/src/main/java/io/servicecomb/core/transport/AbstractTransport.java [hunk 2]
```diff
@@ -50,25 +51,13 @@
   public static final String ENDPOINT_KEY = "cse.endpoint";
 
   private static final long DEFAULT_TIMEOUT_MILLIS = 30000;
+  
+  private static final String DEFAULT_TIMEOUT_KEY = "cse.request.timeout";
```
Comment (WillemJiang) [inline core/src/main/java/io/servicecomb/core/transport/AbstractTransport.java]: Please change the constant to REQUEST_TIMEOUT_KEY.

File: transports/transport-highway/src/test/java/io/servicecomb/transport/highway/TestHighwayClient.java [hunk 3]
```diff
@@ -61,7 +67,22 @@
   Endpoint endpoint = Mockito.mock(Endpoint.class);
 
   Executor excutor = Mockito.mock(Executor.class);
+  
+  @BeforeClass
+  public static void beforeCls() {
+    ConfigUtil.installDynamicConfig();
+    AbstractConfiguration configuration =
+        (AbstractConfiguration) DynamicPropertyFactory.getBackingConfigurationSource();
+    configuration.addProperty("cse.request.timeout", 2000);
```
Comment (WillemJiang) [inline transports/transport-highway/src/test/java/io/servicecomb/transport/highway/TestHighwayClient.java]: Please using the constant value like REQUEST_TIMEOUT_KEY here.

File: core/src/main/java/io/servicecomb/core/transport/AbstractTransport.java [hunk 4]
```diff
@@ -50,25 +51,13 @@
...
-        .getLongProperty("cse.request.timeout", DEFAULT_TIMEOUT_MILLIS)
-        .get();
-    if (msTimeout <= 0) {
-      msTimeout = DEFAULT_TIMEOUT_MILLIS;
-    }
-
-    msRequestTimeout = msTimeout;
-    return msRequestTimeout;
+  public static DynamicLongProperty getRequestTimeoutProperty(){
+    return DynamicPropertyFactory.getInstance().getLongProperty(REQUEST_TIMEOUT_KEY, DEFAULT_TIMEOUT_MILLIS);
```
Comment (wujimin) [inline core/src/main/java/io/servicecomb/core/transport/AbstractTransport.java]: no need to create the property every time.

**Comment by WillemJiang** [comment]
@laijianbin  There are two PR for the same issue, please close the PR which you don't want us to merge.

**Comment by laijianbin** [comment]
no use

**Pull request closed without merging**